website/docs/reference/widgets/table/inline-editing.md

```diff
@@ -235,13 +236,118 @@ This property contains the details of the row that triggered the action (`onSubm
 }
 ```
 
-## Events
+### Events
 
 | Events        | Description                                                                                                            |
 | ------------- | ---------------------------------------------------------------------------------------------------------------------- |
 | **onSubmit**  | Action that gets triggered when the user moves away from editing a cell by pressing the enter key or clicking outside. |
-| **onSave**    | Action that gets triggered when the user clicks the save button.                                                       |
-| **onDiscard** | Action that gets triggered when user clicks discard button                                                             |
+| **onSave**    | Action that gets triggered when the user clicks the save button for a cell or a new row.                                                       |
+| **onDiscard** | Action that gets triggered when user clicks discard button for a cell or a new row.                                                       |
+
+## Adding new rows
+
+In addition to [editing individual cells](#how-to-edit-a-cell), you may allow users to use the Table UI to submit new rows of data.
+
+### Properties
+
+The following properties are related to adding new rows:
+
+| Property              | Type        | Definition               | Code Snippet        |
+| --------------------- | ----------- | ------------------------ | ------------------- |
+| **Allow adding a row** | Widget | Toggles a button in the table which allows users to submit new rows of data. Only columns marked as **Editable** can accept user input. Use the **onSave** event to update the source of the table's data and reflect the user's changes. |
+| **Default Values** | Widget | The values to automatically populate the new row with when a user begins creating a new row. Expects an object with the same keys as the columns in the existing table data. |
+| **isAddRowInProgress** | Binding | Indicates whether a new row is currently being added by the user. | `Table1.isAddRowInProgress` |
+| **newRow** | Binding | This variable contains a reference to the new row object added by the user. | `Table1.newRow` |
+| **isNewRow** | Validation | When writing code for validation properties, this variable indicates whether the cells being validated are part of a new or an existing row. | `isNewRow` |
+
+The subheadings below describe these properties in more detail:
+
+#### Allow adding a row
+
+Turning this on allows users to click a button on the table to start a new row in the table. Users can only input data for columns that are marked as **Editable** in the table properties.
+
+Use the **onSave** event to update the source of the table's data and reflect the user's changes. For example, the **onSave** event might be set to trigger a query that sends the data from a new table row to the database that supplies the table with its values.
```

Comment:
I think instead of "update the source of the table's data", can we write  "update the original datasource table to reflect the user's changes"?